specification/azsadmin/resource-manager/fabric/Microsoft.Fabric.Admin/preview/2019-05-01/NasCluster.json

```diff
@@ -0,0 +1,207 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "description": "Nas Cluster operation endpoints and objects.",
+    "title": "FabricAdminClient",
+    "version": "2019-05-01"
+  },
+  "host": "adminmanagement.local.azurestack.external",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Fabric.Admin/fabricLocations/{location}/nasClusters/{nasCluster}": {
+      "get": {
+        "x-ms-examples": {
+          "Return the requested a nas cluster.": {
+            "$ref": "./examples/NasCluster/Get.json"
+          }
+        },
+        "tags": [
+          "NasClusters"
+        ],
+        "description": "Return the requested nas cluster.",
+        "operationId": "NasClusters_Get",
+        "parameters": [
+          {
+            "$ref": "../2016-05-01/Fabric.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../2016-05-01/Fabric.json#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "../2016-05-01/Fabric.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "#/parameters/NasClusterParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/NasCluster"
+            }
+          },
+          "404": {
```

Comment:
Based on my knowledge, default should be enough. We can wait for ARM's confirm.

Reply:
Normally ARM team would not review AzureStack's spec. This also keep consistent with other spec

Reply:
The question here is whether to add 404  or not?

Reply:
I copy from other spec, if the default behavior is 404 then we don't need to add it

Reply:
No, the normal pattern is to simply have a `default` response listed to deal with all the error.

---
In reply to: [378253937](https://github.com/Azure/azure-rest-api-specs/pull/8339#discussion_r378253937) [](ancestors = 378253937)

Reply:
Replace with following:

,

          "default": {

            "description": "*** Error Responses: ***\n\n * 400 Bad Request.\n\n * 500 Internal Server Error."

          }

Reply:
@majastrz in many of the azure stack specs, i see that we explicitly define 404. Is this not a general practice? I remember if we dont , the autorest generated SDK wont throw the exeption or something like that?

Reply:
It is rare in ARM RPs. For anything caught by the `default` response, the .net SDK will throw an exception. If you have a reason to not throw, you can declare it.

Reply:
I believe we did that, because , we only ship powershell for these administrator APIs and in general Get-* commands are supposed to return nothing for the non existing  items rather than throwing exception. This is  a general powershell guidleline

Reply:
Its OK to have this default for now. Lets see the behavior when we generate the powershell and we can decide to make another PR if needed

Reply:
The SDK will throw an exception, but you can always catch it in your cmdlets. This choice will also impact all SDKs you generate (.net, Python, JS, Ruby, PHP, etc.) The SDK should not be generated/designed for one consumer (Azure PowerShell).

Reply:
sure. that makes sense